### ISSUE TITLE
May break on Ruby versions below `3.2.0` due to non-evaluatable `Symbol#inspect`

### ISSUE THREAD

**Issue by prikha** [mbj/unparser#359]
For `String` and `Symbol` Unparser [uses](https://github.com/mbj/unparser/blob/main/lib/unparser/emitter/primitive.rb#L18) `#inspect` for serialization back to Ruby-code. Which makes sense. But apparently `#inspect` hasn't always worked the same way for Symbols across Ruby versions:

```
ruby-2.0.0-p0       :"8 >="
...
ruby-2.2.0-preview1 :"8 >="
ruby-2.2.0-preview2 :8 >=
...
ruby-3.2.0-preview1 :8 >=
ruby-3.2.0-preview2 :"8 >="
...
ruby-3.3.0          :"8 >="
```

This bug has been reported in Ruby itself and fixed in >3.2.0 
https://bugs.ruby-lang.org/issues/18905

There is nothing to do about it, but might be of interest for those still running older Rubies.

**Comment by mbj** [comment]
Context at the time I had 2.7 still in the text matrix, I was round tripping rubyspec (on CI) that includes this kinds of symbols.

**Comment by mbj** [comment]
So 2.7 does not compile on a bison symbol error, and I do not actually want to invest time into an EOL ruby. If someone can proof me 2.7 is indeed broken for the 0.6.12 releases, I'm happy to yank these gems and or put a warning into this repos readme.

**Comment by mbj** [comment]
@prikha or are you saying its borken on 3.x? Can you give me something that produces an issue on `bundle exec bin/unparser -e '$broken'` ?

**Comment by prikha** [comment]
Sorry Markus, it may sounded a bit misleading. There is no action point, I just wanted to pinpoint that older versions of `unparser` are affected by the bug mentioned.

####`ruby 3.2.0`: 
```sh
./bin/unparser -e ':"8 <="'
Success: (string)
```

#### `ruby 2.7.8`: 
```
./bin/unparser -e ':"8 <="'
(string)
Original-Source:
:"8 <="
Generated-Source:
:8 <=
Original-Node:
(sym :8 <=)
Generated-Node:
#<Parser::SyntaxError: unexpected token tCOLON>
...
Error: (string)
```

With Ruby '>ruby-3.2.0-preview2` it's solved, and gemspec specifies dependency well.
If anyone would face an issue, they'll probably come by this ticket. Feel free to close it.

**Comment by prikha** [comment]
Again I didn't mean that you have to do any changes or spend some extra time, you're doing great running this project already, cheers! 🍺

**Comment by prikha** [comment]
The only thing I would to is to update the required ruby version to `>3.2` in the gemspec, so that the fix is present in the Ruby itself.

**Comment by mbj** [comment]
Nah, I need `>= 3.0` if its borken on any ruby `>=3.0` I'll work around it for these versions.

**Comment by prikha** [comment]
@mbj may I ask for a tip - there is a fix for 3.0, 3.1 ☝🏻 but `Mutant` is catching me by the hand with ruby version check and I'm a bit puzzled how should we overcome that. E.g. if/else branches are equivalent on 3.3, 3.2 so mutating if condition won't fail any test in CI as it's the sole purpose of the fix to produce same results on different versions.

**Comment by mbj** [comment]
@prikha Yeah so mutant needs ruby version based disable. What I'd do right now is to:

* mutation cover it on 3.0, and 3.1
* add a `# mutant:disable` to the method, with a comment about its version specific.

Ideally I get mutant to support version specific disables in the future.

**Comment by mbj** [comment]
@prikha your PR drops support for non EOL rubies, not acceptable. This needs a ruby version specific branch to add the missing quotes, if any.

**Comment by prikha** [comment]
@mbj ok closed the first one, and updated the second one. Mind taking a look 🙏🏻

**Comment by mbj** [comment]
@prikha Sorry travelling right now. I cannot offer determinism ATM.

**Comment by mbj** [comment]
@prikha Released as 0.6.13, ty!
